lib/src/commit_builder.rs

```diff
@@ -85,6 +85,15 @@ impl CommitBuilder<'_> {
         {
             commit.author.email = commit.committer.email.clone();
         }
+
+        // Reset timestamp on discardable commits if author is committer
+        if commit.author.name == commit.committer.name
+            && commit.author.email == commit.committer.email
```

Comment:
(Optional) I wonder whether we should mention the cases when we'd expect checking the author name/email to matter.

One case I can think of are merge commit, as discussed by Martin and Yuya just below. In repos that use merge commits to merge PRs, the maintainer might create empty merge commits, and the date of those commits should not be changed.

Reply:
I don't personally care for this feature at all, but if we have it (which I'm fine with), I think it should be consistent between merge commits and other commits. Why do you think some merge commits (even without description and changes) should keep their initial timestamp when they get rebased etc?

Reply:
Oh, maybe you're discussing something else than I thought. Are you asking why we don't want to update the author timestamp if we rewrite and empty commit created by someone else? I think it make sense to avoid updating the author timestamp in those cases. If you reset the author to be you, then I think the timestamp should change, however.

Reply:
I updated the comment to try to make my reasoning more clear and provide an example.

Reply:
I was indeed a little mistaken; the merge commits I was talking about would have a non-empty description.

Indeed, I'm not sure what the situation would be where it matters whether we update the timestamps of discardable commits authored by somebody else. The one example I thought I came up with in my comment above isn't actually an example because of non-empty description.

I don't think it's a big deal either way, it's fine to keep things as they are. Indeed, it's a good thing that resetting the author to be you updates the timestamp.

It just bothers me a little to have a piece of logic I can't justify with an example. If we *can* come up with an example why this restriction is a good idea (or why it is a bad idea, for that matter), I'd want to write it down.

Reply:
Perhaps we should just say:

>  Reset author timestamp on discardable commits, if the author is the
committer. While it's unlikely we'll have somebody else's commit with no description in our repo, we'd like to be extra-safe.

Another point I just thought of: the most likely situation this will make a difference is if a user uses two email addresses for themselves, in which case they might be confused why the timestamp on a commit they think is theirs isn't updated. However, I don't think that's a big reason to slow down this PR; we can change it later if it becomes a problem.